kolibri/core/assets/src/views/immersive-app-bar/index.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+
+  <ui-toolbar
+    :title="appBarTitle"
+    textColor="white"
+    type="colored"
+    :navIcon="icon"
+    :showIcon="showIcon"
+    @nav-icon-click="$emit('navIconClick')"
+    :style="{ height: height + 'px' }"
+  >
+    <div v-if="hasRoute" slot="icon">
+      <router-link :to="route">
+        <!-- TODO add aria label? -->
+        <ui-icon-button
+          type="flat"
+          @click="$emit('navIconClick')"
+          :icon="icon"
+          class="icon"
+        />
+      </router-link>
+
+    </div>
+  </ui-toolbar>
+
+</template>
+
+
+<script>
+
+  import uiToolbar from 'keen-ui/src/UiToolbar';
+  import uiIconButton from 'keen-ui/src/UiIconButton';
+  import { validateLinkObject } from 'kolibri.utils.validators';
+
+  export default {
+    name: 'immersiveAppBar',
+    components: {
+      uiToolbar,
+      uiIconButton,
+    },
+    props: {
+      appBarTitle: {
```

Comment:
Can just name this prop `title`

Reply:
This is carryover from the last version of `core-base`, though I'm worried it'll still cause some confusion.

Currently, we've got 3 different "titles" at the "core" level.

- [The title in store, used to set the `document.title`](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/state/store.js#L83)
- [The title used by `app-bar`](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/views/core-base.vue#L125)
- [`PageName`, commonly used to name our plugin subpages](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/coach/assets/src/state/actions/lessons.js#L29)

I know that last one doesn't include title, but I thought maintaining the artificial scope by using `AppBarTitle` would be best. You disagree?

Reply:
True, but this immersive-app-bar component is standalone so it should not really be affected by these other titles? `<immersive-app-bar :title="appBarTitle" >` should be good enough to avoid confusion, right?

Reply:
I guess since the document title is set via mutation it might be less of a big deal.

This change would also lead to a pretty wide refactor, given that it's a carryover from core-base. One of my intentions in this refactor was to maintain the changes as small in scope as possible.